#!/bin/sh -e
## 01_makefile.dpatch.dpatch by Otavio Salvador <otavio@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fix problems with autobuilders (#178762, #198519)

if [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch}"

case "$1" in
       -patch) patch $patch_opts -p1 < $0;;
       -unpatch) patch $patch_opts -p1 -R < $0;;
        *)
                echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
                exit 1;;
esac

exit 0

@DPATCH@
diff -urNad /home/otavio/devel/debian/tcm/tcm-2.20/src/gl/Makefile tcm-2.20/src/gl/Makefile
--- /home/otavio/devel/debian/tcm/tcm-2.20/src/gl/Makefile	2003-06-23 16:51:59.000000000 -0300
+++ tcm-2.20/src/gl/Makefile	2003-06-23 16:52:01.000000000 -0300
@@ -56,8 +56,7 @@
 clean: xclean
 
 xclean::
-	$(RM) instances.h
-	touch instances.h 
+	echo > instances.h
 
 depend: xdepend
 
diff -urNad /home/otavio/devel/debian/tcm/tcm-2.20/src/Makefile.gcc tcm-2.20/src/Makefile.gcc
--- /home/otavio/devel/debian/tcm/tcm-2.20/src/Makefile.gcc	2003-01-09 12:52:41.000000000 -0200
+++ tcm-2.20/src/Makefile.gcc	2003-06-23 16:53:15.000000000 -0300
@@ -17,13 +17,13 @@
 semidynamiclibs: libglobal.a libgui.so libeditor.so libdiagram.so libtable.so
 
 tcm$(EXEEXTENSION):
-	( cp ed/edinstances.h gl/instances.h )
+	( cat ed/edinstances.h > gl/instances.h )
 	( cd ./gl ; touch llist.c )
 	( $(MAKE) libglobal.a )
 	( cd ./ed ; $(MAKE) $@)
 
 tcmd$(EXEEXTENSION):
-	( cp sd/sdinstances.h gl/instances.h )
+	( cat sd/sdinstances.h > gl/instances.h )
 	( cd ./gl ; touch llist.c )
 	( $(MAKE) libglobal.a )
 	( cd ./sd ; $(MAKE) $@)
@@ -36,31 +36,31 @@
 	tsnd$(EXEEXTENSION) trpg$(EXEEXTENSION) tscd$(EXEEXTENSION) \
 	tsqd$(EXEEXTENSION) tcbd$(EXEEXTENSION) tcpd$(EXEEXTENSION) \
 	tdpd$(EXEEXTENSION):
-	( cp dg/dginstances.h gl/instances.h )
+	( cat dg/dginstances.h > gl/instances.h )
 	( cd ./gl ; touch llist.c )
 	( $(MAKE) libglobal.a )
 	( cd ./sd ; $(MAKE) $@)
 
 tcmfv$(EXEEXTENSION) tdfd$(EXEEXTENSION) tefd$(EXEEXTENSION):
-	( cp sd/fv/dfinstances.h gl/instances.h )
+	( cat sd/fv/dfinstances.h > gl/instances.h )
 	( cd ./gl ; touch llist.c )
 	( $(MAKE) libglobal.a )
 	( cd ./sd ; $(MAKE) $@)
 
 tcmt$(EXEEXTENSION) tgt$(EXEEXTENSION) ttdt$(EXEEXTENSION) ttut$(EXEEXTENSION) tfet$(EXEEXTENSION):
-	( cp tb/tbinstances.h gl/instances.h )
+	( cat tb/tbinstances.h > gl/instances.h )
 	( cd ./gl ; touch llist.c )
 	( $(MAKE) libglobal.a )
 	( cd ./st ; $(MAKE) $@)
 
 # csod:
-# 	( cp cx/cx_instances.h gl/instances.h )
+# 	( cat cx/cx_instances.h > gl/instances.h )
 # 	( cd ./gl ; touch llist.c )
 # 	( $(MAKE) libglobal.a )
 # 	( cd ./cx ; $(MAKE) $@)
 
 # ced:
-# 	( cp dg/dginstances.h gl/instances.h )
+# 	( cat dg/dginstances.h > gl/instances.h )
 # 	( cd ./gl ; touch llist.c )
 # 	( $(MAKE) libglobal.a )
 # 	( cd ./cx ; $(MAKE) $@)
diff -urNad /home/otavio/devel/debian/tcm/tcm-2.20/src/Makefile.GEN tcm-2.20/src/Makefile.GEN
--- /home/otavio/devel/debian/tcm/tcm-2.20/src/Makefile.GEN	2003-01-09 12:52:41.000000000 -0200
+++ tcm-2.20/src/Makefile.GEN	2003-06-23 16:54:12.000000000 -0300
@@ -72,14 +72,14 @@
 # the touchy thing is necessary that the right template
 # instances are linked in the static lib.
 libdiagram.a:
-	( cp dg/dginstances.h gl/instances.h )
+	( cat dg/dginstances.h > gl/instances.h )
 	( cd ./gl ; touch llist.c )
 	( $(MAKE) libglobal.a )
 	( cd ./dg ; $(MAKE) libdiagram.a)
 	( cd $(LIB_DIR) ; $(RANLIB) libdiagram.a)
 
 libtable.a:
-	( cp tb/tbinstances.h gl/instances.h )
+	( cat tb/tbinstances.h > gl/instances.h )
 	( cd ./gl ; touch llist.c )
 	( $(MAKE) libglobal.a )
 	( cd ./tb ; $(MAKE) libtable.a)
